math/mode.go

```diff
@@ -18,32 +18,34 @@ var ErrEmptySlice = errors.New("empty slice provided")
 
 func Mode[T constraints.Number](numbers []T) (T, error) {
 
+	countMap := make(map[T]int)
+
 	n := len(numbers)
 
 	if n == 0 {
 		return 0, ErrEmptySlice
 	}
 
-	maxCount := 0
-	var maxValue T = 0
-
 	for i := 0; i < n; i++ {
-
-		count := 0
-
-		for k := 0; k < n; k++ {
-
-			if numbers[k] == numbers[i] {
-				count++
-			}
+		temp := numbers[i]
+		value, check := countMap[temp]
+		if check == false {
+			countMap[temp] = 1
+		} else {
+			countMap[temp] = value + 1
 		}
+	}
+
+	var mod T
```

Comment:
You can call this `mode` instead of `mod` - I believe you wrote it as `mod` because the package name was conflicting with it earlier but now it should be fine to call it `mode`

Reply:
Please don't mark things as resolved without a comment or without addressing the suggestion presented.